File: OtherSource/AlertViewHelper.m
//
//  AlertViewHelper.m
//  AVOSDemo
//
//  Created by lzw on 15/5/21.
//  Copyright (c) 2015年 AVOS. All rights reserved.
//

#import "AlertViewHelper.h"

@interface AlertViewHelper()<UIAlertViewDelegate>

@property (nonatomic, copy) InputAlertViewFinishBlock finishBlock;

@property (nonatomic, strong) NSString *inputText;

@end

@implementation AlertViewHelper

- (void)showInputAlertViewWithMessage:(NSString*)message block:(InputAlertViewFinishBlock)block {
    self.finishBlock = block;
    UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:nil message:message delegate:self cancelButtonTitle:@"取消" otherButtonTitles:@"确定", nil];
    alertView.alertViewStyle = UIAlertViewStylePlainTextInput;
    [alertView show];
}

- (void)alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)buttonIndex {
    if (buttonIndex == 1) {
        self.inputText = [alertView textFieldAtIndex:0].text;
    }
}

- (void)alertView:(UIAlertView *)alertView didDismissWithButtonIndex:(NSInteger)buttonIndex {
    if (self.finishBlock) {
        self.finishBlock(self.inputText);
    }
    self.inputText = nil;
    self.finishBlock = nil;
}

@end

File: LeanStorageDemoSwift/LeanStorageDemoSwift/AlertViewHelper.swift
//
//  AlertViewHelper.swift
//  LeanStorageDemoSwift
//
//  Created by lzw on 15/9/3.
//  Copyright (c) 2015年 LeanCloud All rights reserved.
//

import UIKit

class AlertViewHelper:NSObject, UIAlertViewDelegate {
    
    var finishBlock: ((input:String?) -> Void)?
    
    func showInputView(message: String, block:(input:String?) -> Void) {
        finishBlock = block
        let alertView = UIAlertView(title: nil, message: message, delegate: self as UIAlertViewDelegate, cancelButtonTitle: "取消")
        alertView.addButtonWithTitle("确定")
        alertView.alertViewStyle = UIAlertViewStyle.PlainTextInput
        alertView.show()
    }
    
    func alertView(alertView: UIAlertView, clickedButtonAtIndex buttonIndex: Int) {
        if buttonIndex == 1 {
            if finishBlock != nil {
                finishBlock!(input:alertView.textFieldAtIndex(0)?.text)
            }
        }
    }
}
